### PR TITLE
Added accounts and login/logout for PCA

### PR DESCRIPTION
**Pull request by dlike230**
Also added a folder of `shared-components` for components that are shared between PCA and PCP. See updated README for setup details.

### CODE REVIEW

**Comment by sigmachirality** [comment]
I'm worried that PCP's jank CSS will contaminate the PCA codebase. I like styled components, can we just make all the shared components styled components so the css scoping is contained to the components?

**Comment by dlike230** [comment]
> I'm worried that PCP's jank CSS will contaminate the PCA codebase. I like styled components, can we just make all the shared components styled components so the css scoping is contained to the components?

Yeah good idea. The only thing we can't get rid of as easily is Bulma.

**Comment by dlike230** [comment]
All functionality here is done now -- just need to do the refactoring.

File: frontend/plan/Dockerfile [hunk 1]
```diff
@@ -9,6 +9,7 @@ COPY package*.json /app/
 RUN npm install
 
 COPY . /app/
+COPY ../shared-components/src /app/src/components
```
Comment (davish) [inline frontend/plan/Dockerfile:12]: Just tested a docker build locally and this doesn't work... will need to find a workaround. Probably eventually we should make this an NPM package that we publish as part of the build step?

Comment (dlike230) [reply frontend/plan/Dockerfile:12]: It doesn't need to be a published npm package, we can just put a github link in our package.json. Also we could also use git submodules as another alternative.

**Pull request merged**